core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -486,6 +486,17 @@ default void renameColumn(ConnectorSession session, ConnectorTableHandle tableHa
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming columns");
     }
 
+    /**
+     * Add the specified field, potentially nested, to a row.
+     *
+     * @param fieldPath path to a field within the column, including leaf field name.
+     */
+    @Experimental(eta = "2023-09-01") // TODO add support for rows inside arrays and maps and for anonymous row fields
+    default void renameField(ConnectorSession session, ConnectorTableHandle tableHandle, List<String> fieldPath, String target)
```

Comment:
document what `target` is
in particular document that we don't expect field to move around